src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs

```diff
@@ -512,11 +512,12 @@ public bool Exists(Predicate<T> match)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.match);
             }
 
-            for (int i = 0; i < _size; i++)
+            ReadOnlySpan<T> span = new ReadOnlySpan<T>(_items);
+            foreach (T element in span)
             {
-                if (match(_items[i]))
+                if (match(element))
```

Comment:
This is actually a change in behavior. If the delegate were, for example, to add an element to the list, previously that new element would be evaluated as part of the for loop. Now it'll be missed, because the span being iterated through won't include it.

Reply:
Right, I see what you mean. My change in using the Foreach loop instead of the For was actually to avoid an IndexOutOfRangeException (even if someone else on another thread is removing values). But I hadn't thought about this case where you play with the list inside the predicate. 🥲
But now I'm wondering: how can this work in the Linq method? The method extracts the underlying array (or list) with “Unsafe.As” and then uses the span in the foreach loop in the same way. Wouldn't this create the same problem? Thank you for your response by the way :)